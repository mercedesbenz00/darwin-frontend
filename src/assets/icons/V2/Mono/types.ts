export enum IconVariant {
  DEFAULT = 'default',
  DISABLED = 'disabled',
  SELECTED = 'selected',
  NEGATIVE = 'negative',
}
