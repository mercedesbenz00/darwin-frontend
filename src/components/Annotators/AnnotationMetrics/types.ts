import { MembershipPayload, DatasetAnnotationReportPayload } from '@/store/types'

export type AnnotationDataRange = 'total' | 'month' | 'week' | 'day'
export type AnnotationDataGranularity = 'day' | 'hour'

export type ChartDimension =
  'annotationTime' |
  'imagesAnnotated' |
  'avgTimePerAnnotation' |
  'annotationsCreated' |
  'reviewPassRate'

export type AdaptedDataPoint =
  {[k in Exclude<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number } &
  {[k in Extract<ChartDimension, 'reviewPassRate' | 'avgTimePerAnnotation'>]: number | null} &
  { userId: number | null, date: string }

export type AggregatedDataPoint = Pick<
    DatasetAnnotationReportPayload,
    | 'annotation_time'
    | 'annotations_approved'
    | 'annotations_created'
    | 'date'
    | 'images_annotated'
    | 'images_approved'
    | 'images_rejected'
    | 'user_id'
  >

export type AnnotationDimensionData = {
  annotationTime: number,
  annotationsCreated: number,
  avgTimePerAnnotation: number | null,
  imagesAnnotated: number,
  reviewPassRate: number | null
}

export type AnnotationChartData = {
  data: AdaptedDataPoint[]
  dimensions: AnnotationDimensionData
  member?: MembershipPayload
  visible: boolean
}

export type AnnotationBarChartData = {
  data: AggregatedDataPoint[]
  member?: MembershipPayload
}

/**
 * Format for a single item to be shown in a chart tooltip.
 *
 * A tooltip will contain multiple items
 */
export type ChartTooltipData = { member: MembershipPayload | null, value: string}
