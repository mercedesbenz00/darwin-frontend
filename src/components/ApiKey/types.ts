export type AbilityGroup =
  'manage_comments' |
  'manage_tasks' |
  'manage_datasets' |
  'manage_team' |
  'manage_team_members' |
  'manage_models'

export type Group = { id: AbilityGroup; name: string }
export type GroupOption = Group & { selected: boolean }

export enum SupportedLanguages {
  shell = 'shell',
  python = 'python',
  javascript = 'javascript',
  elixir = 'elixir',
  cli = 'cli'
}

export type AccessCodeExample = { language: SupportedLanguages, code: string }
