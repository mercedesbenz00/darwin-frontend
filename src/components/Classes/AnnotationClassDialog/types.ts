import { AnnotationClassPayload, AnnotationClassMetadata, AnnotationTypeName } from '@/store/types'

type PendingFields =
  | 'crop_key'
  | 'crop_url'
  | 'image_height'
  | 'image_width'
  | 'key'
  | 'original_image_url'
  | 'scale'
  | 'x'
  | 'y'

/**
 * Holds a class image that isn't necessarily persisted to the backend yet.
 * That means it might be missing any of the url fields
 *
 * It can be saved to backend once at least the `key` field is set, as that one
 * can be used to infer/compute all the other fields.
 *
 * - `crop_key` - repleace "original" with "crop" in `key` field
 * - `crop_url` - signed s3 url (by backend), for `crop_key`
 * - `original_image_url` - signed s3 url (by backend) for `key`
 */
export type PendingClassImage =
  Omit<AnnotationClassPayload['images'][0], PendingFields>
  & Partial<Pick<AnnotationClassPayload['images'][0], PendingFields>>

// eslint-disable-next-line camelcase
export type PendingClassImageWithUrlSet = PendingClassImage & { original_image_url: string }

export const isClassImage = (
  image: PendingClassImage
): image is AnnotationClassPayload['images'][0] =>
  !!image.key && !!image.crop_key

export type AnnotationClassData = {
  datasets: AnnotationClassPayload['datasets']
  description: string | null
  hotkey: null | string
  id: number | null
  images: PendingClassImage[]
  metadata: AnnotationClassMetadata
  name: string | null
}

export type DefaultAnnotationClassData = Partial<AnnotationClassData> & {
  selectedMainAnnotationType?: AnnotationTypeName
}

export type AnnotationClassValidationErrors = {
  annotationTypes?: string
  name?: string
  skeleton?: string
}

export const VIEWPORT_WIDTH: number = 150
export const VIEWPORT_HEIGHT: number = 150

export type CropInfo = {
  orientation: number
  points: [string, string, string, string]
  zoom: number
}

/**
 * VueCroppie is really archaic and requires us to get crop info via $ref,
 * so this type defines that ref
 */
export type Croppie = {
  /** Binds options to the underlying croppie library */
  bind: (options: Pick<CropInfo, 'points' | 'zoom'> & { url: string}) => void

  /** Retrieves current crop info */
  get: () => CropInfo
  result: (options?: Record<string, string>, callback?: Function) => void

  /** Sets scale/zoom level of cropper */
  setZoom: (value: number) => void
}
