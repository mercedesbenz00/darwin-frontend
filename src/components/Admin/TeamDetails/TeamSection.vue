<template>
  <div
    class="section"
    :class="{'section--inactive': !active}"
  >
    <div class="section__title">
      <slot name="title" />
    </div>
    <div class="section__body">
      <slot name="body" />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component({
  name: 'team-section'
})
export default class TeamSection extends Vue {
  @Prop({ required: false, default: true, type: Boolean as () => boolean })
  active!: boolean
}
</script>

<style lang="scss" scoped>
.section {
  @include row;
  flex-wrap: wrap;
  margin-bottom: 25px;
  padding: 20px;
  background: $colorSecondaryLight3;
  border-radius: 3px;

}

.section .section--inactive {
  opacity: .3;
}

.section .section__title {
  width: 80%;
  margin-right: 20%;
  @include typography(xl, headlines, bold);
  margin-bottom: 10px;
}

.section .section__body {
  @include row;
  flex-wrap: wrap;

  gap: 15px;
}
</style>
