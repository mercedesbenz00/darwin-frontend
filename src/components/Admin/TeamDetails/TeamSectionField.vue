<template>
  <div
    class="field"
    :class="{ 'field--full': fullWidth }"
  >
    <label class="field__label">
      <slot name="label" />
    </label>
    <div class="field__value">
      <slot name="value" />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component({
  name: 'team-section-field'
})
export default class TeamSectionField extends Vue {
  @Prop({ required: false, default: false, type: Boolean as () => boolean })
  fullWidth!: boolean
}
</script>

<style lang="scss" scoped>
.field {
  width: 300px;
  @include col;
}

.field.field--full {
  width: 100%;
}

.field.field__label {
  font-weight: bold;
  margin-bottom: 10px;
}
</style>
