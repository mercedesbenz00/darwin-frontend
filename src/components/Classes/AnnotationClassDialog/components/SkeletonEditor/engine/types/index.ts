export * from './SkeletonEdgeType'
export * from './SkeletonNodeType'
