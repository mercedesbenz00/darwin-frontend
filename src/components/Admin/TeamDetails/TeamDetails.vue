<template>
  <div class="team">
    <h1 class="team__title">
      {{ team.name }} team information
    </h1>
    <div class="team__actions">
      <positive-button @click="joinTeam">
        Add me to this team
      </positive-button>
      <primary-button
        v-if="!team.features.includes('MODEL_STAGE')"
        @click="enableFeature('MODEL_STAGE')"
      >
        Enable AI Stages
      </primary-button>
      <primary-button
        v-if="!team.features.includes('CODE_STAGE')"
        @click="enableFeature('CODE_STAGE')"
      >
        Enable Code Stages
      </primary-button>
      <primary-button
        v-if="!team.features.includes('BLIND_STAGE')"
        @click="enableFeature('BLIND_STAGE')"
      >
        Enable Blind Stages
      </primary-button>
      <primary-button
        v-if="!team.features.includes('VIDEO_STREAM')"
        @click="enableFeature('VIDEO_STREAM')"
      >
        Enable Video Stream
      </primary-button>
      <primary-button
        v-if="!team.features.includes('MODEL_TOOL')"
        @click="enableFeature('MODEL_TOOL')"
      >
        Enable Model Tool
      </primary-button>
    </div>
    <team-section>
      <template #body>
        <team-section-field>
          <template #label>
            Owner
          </template>
          <template #value />
        </team-section-field>
        <team-section-field>
          <template #label>
            Creation Date
          </template>
          <template #value>
            {{ creationDate }}
          </template>
        </team-section-field>
      </template>
    </team-section>
    <team-section>
      <template #title>
        Note
      </template>
      <template #body>
        <team-section-field>
          <template #label />
          <template #value>
            {{ team.note }}
          </template>
        </team-section-field>
      </template>
    </team-section>
    <stripe :team="team" />
    <credits-v3 :team="team" />
    <storage-v3 :team="team" />
    <partner-status :team="team" />
    <team-section>
      <template #title>
        {{ team.user_count }} users
      </template>
      <template #body>
        User list in development
      </template>
    </team-section>
    <team-section>
      <template #title>
        {{ team.dataset_count }} datasets
      </template>
      <template #body>
        Dataset list in development
      </template>
    </team-section>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'

import { createTeamFeature } from '@/store/modules/admin/actions/createTeamFeature'
import { TeamPayload } from '@/store/modules/admin/types'
import { FeaturePayload, StoreActionPayload } from '@/store/types'
import { formatDate, getFullName } from '@/utils'

import CreditsV3 from './CreditsV3.vue'
import PartnerStatus from './PartnerStatus.vue'
import StorageV3 from './StorageV3.vue'
import Stripe from './Stripe.vue'
import TeamSection from './TeamSection.vue'
import TeamSectionField from './TeamSectionField.vue'

@Component({
  name: 'team-details',
  components: { CreditsV3, PartnerStatus, StorageV3, Stripe, TeamSection, TeamSectionField }
})
export default class TeamDetails extends Vue {
  @Prop({ required: true, type: Object })
  team!: TeamPayload

  get ownerName (): string {
    return getFullName({
      first_name: this.team.owner_first_name,
      last_name: this.team.owner_last_name
    })
  }

  get ownerEmail (): string {
    return this.team.owner_email
  }

  get creationDate (): string {
    return formatDate(this.team.creation_date, 'MMMM Do, YYYY HH:mm:ss')
  }

  async joinTeam (): Promise<void> {
    const { error } = await this.$store.dispatch('admin/joinTeam', { id: this.team.id })

    if (error) {
      this.$store.dispatch('toast/warning', {
        content: `Could not join ${this.team.name}. Please, refresh and try again.`
      })
      return
    }

    this.$store.dispatch('toast/notify', {
      content: `You've successfully joined ${this.team.name}`
    })
  }

  async enableFeature (feature: FeaturePayload['name']): Promise<void> {
    const { id, name } = this.team
    const payload: StoreActionPayload<typeof createTeamFeature> = {
      teamId: id,
      feature
    }

    const { error } = await this.$store.dispatch('admin/createTeamFeature', payload)

    if (error) {
      const content = `Couldn't enable ${feature} for team ${name}. Please refresh and try again.`
      this.$store.dispatch('toast/warning', { content })
      return
    }

    const content = `You've successfully enabled ${feature} for team ${name}`
    this.$store.dispatch('toast/notify', { content })
  }
}
</script>

<!-- eslint-disable vue-scoped-css/enforce-style-type -->
<style lang="scss">
.team {
  background: $colorWhite;
  border-radius: 3px;
  padding: 25px;
  width: auto;
}

.team__title {
  @include typography(xxl, headlines, bold);
  margin-bottom: 25px;
}

.team__actions {
  @include row;
  margin-bottom: 25px;
}

.team__actions a, .team__actions button {
  margin-right: 15px;
}

</style>
