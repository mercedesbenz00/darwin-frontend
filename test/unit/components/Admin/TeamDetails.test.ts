import { shallowMount, createLocalVue } from '@vue/test-utils'
import flushPromises from 'flush-promises'
import Vuex from 'vuex'

import createTestStore from 'test/unit/createTestStore'
import { buildAdminTeamPayload } from 'test/unit/factories'

import TeamDetails from '@/components/Admin/TeamDetails/TeamDetails.vue'
import TeamSection from '@/components/Admin/TeamDetails/TeamSection.vue'
import TeamSectionField from '@/components/Admin/TeamDetails/TeamSectionField.vue'
import { installCommonComponents } from '@/plugins/components'
import { StripeSubscriptionStatus } from '@/store/modules/billing/types'

const localVue = createLocalVue()
localVue.use(Vuex)
installCommonComponents(localVue)

let store: ReturnType<typeof createTestStore>
let propsData: { team: ReturnType<typeof buildAdminTeamPayload> }

const stubs = { TeamSection, TeamSectionField }

beforeEach(() => {
  store = createTestStore()
  propsData = {
    team: buildAdminTeamPayload({
      creation_date: '2010-05-01T12:05:55',
      credit_next_expiry: '2040-05-01T14:22:33',
      dataset_count: 5,
      id: 1,
      name: 'V7',
      note: 'Asked for extra credits',
      owner_email: 'joe@v7test.com',
      owner_first_name: 'Joe',
      owner_last_name: 'Owner',
      slug: 'v7',
      stripe_id: 'cus12345678',
      subscription_id: 'sub12345678',
      subscription_period_end: 2524608000,
      subscription_period_start: 2524708000,
      subscription_status: StripeSubscriptionStatus.Cancelled,
      user_count: 10
    })
  }
})

it('matches snapshot', async () => {
  const wrapper = shallowMount(TeamDetails, { localVue, propsData, store, stubs })
  await flushPromises()

  expect(wrapper).toMatchSnapshot()
})

it('dispatches action to enable MODEL_STAGE on click', async () => {
  const wrapper = shallowMount(TeamDetails, { localVue, propsData, store, stubs })
  await flushPromises()

  const button = wrapper
    .findAll('primary-button-stub').wrappers
    .find(b => b.text() === 'Enable AI Stages')

  expect(button).toBeTruthy()
  await button!.vm.$emit('click')

  expect(store.dispatch).toHaveBeenCalledWith('admin/createTeamFeature', {
    feature: 'MODEL_STAGE',
    teamId: propsData.team.id
  })
})

it('dispatches action to enable CODE_STAGE on click', async () => {
  const wrapper = shallowMount(TeamDetails, { localVue, propsData, store, stubs })
  await flushPromises()

  const button = wrapper
    .findAll('primary-button-stub').wrappers
    .find(b => b.text() === 'Enable Code Stages')

  expect(button).toBeTruthy()
  await button!.vm.$emit('click')

  expect(store.dispatch).toHaveBeenCalledWith('admin/createTeamFeature', {
    feature: 'CODE_STAGE',
    teamId: propsData.team.id
  })
})

it('dispatches action to enable BLIND_STAGE on click', async () => {
  const wrapper = shallowMount(TeamDetails, { localVue, propsData, store, stubs })
  await flushPromises()

  const button = wrapper
    .findAll('primary-button-stub').wrappers
    .find(b => b.text() === 'Enable Blind Stages')

  expect(button).toBeTruthy()
  await button!.vm.$emit('click')

  expect(store.dispatch).toHaveBeenCalledWith('admin/createTeamFeature', {
    feature: 'BLIND_STAGE',
    teamId: propsData.team.id
  })
})
