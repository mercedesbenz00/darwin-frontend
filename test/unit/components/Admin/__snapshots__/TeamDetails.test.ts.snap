// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div class="team">
  <h1 class="team__title">
    V7 team information
  </h1>
  <div class="team__actions">
    <positive-button-stub>
      Add me to this team
    </positive-button-stub>
    <primary-button-stub>
      Enable AI Stages
    </primary-button-stub>
    <primary-button-stub>
      Enable Code Stages
    </primary-button-stub>
    <primary-button-stub>
      Enable Blind Stages
    </primary-button-stub>
    <primary-button-stub>
      Enable Video Stream
    </primary-button-stub>
    <primary-button-stub>
      Enable Model Tool
    </primary-button-stub>
  </div>
  <div class="section">
    <div class="section__title"></div>
    <div class="section__body">
      <div class="field"><label class="field__label">
          Owner
        </label>
        <div class="field__value"></div>
      </div>
      <div class="field"><label class="field__label">
          Creation Date
        </label>
        <div class="field__value">
          May 1st, 2010 12:05:55
        </div>
      </div>
    </div>
  </div>
  <div class="section">
    <div class="section__title">
      Note
    </div>
    <div class="section__body">
      <div class="field"><label class="field__label"></label>
        <div class="field__value">
          Asked for extra credits
        </div>
      </div>
    </div>
  </div>
  <stripe-stub team="[object Object]"></stripe-stub>
  <credits-v3-stub team="[object Object]"></credits-v3-stub>
  <storage-v3-stub team="[object Object]"></storage-v3-stub>
  <partner-status-stub team="[object Object]"></partner-status-stub>
  <div class="section">
    <div class="section__title">
      10 users
    </div>
    <div class="section__body">
      User list in development
    </div>
  </div>
  <div class="section">
    <div class="section__title">
      5 datasets
    </div>
    <div class="section__body">
      Dataset list in development
    </div>
  </div>
</div>
`;
